Makefile

```diff
@@ -147,12 +147,12 @@ FFLAGS ?= $(OPT) $(FFLAGS.$(FC_VENDOR))
 ifeq ($(COVERAGE), 1)
   CFLAGS += --coverage
   CXXFLAGS += --coverage
-  LDFLAGS += --coverage
+  override LDFLAGS += --coverage
```

Comment:
@jedbrown, if I understood the documentation correctly, then this `override` needs to follow `LDFLAGS` all over the makefile

Reply:
This looks like it will pass CI, though I want to make sure this has the intended behavior before merging

Reply:
There are many occurrences of `LDFLAGS` in the makefiles. I see that there is also `CEED_LDFLAGS`. Would it make sense to drop this MR and pass `CEED_LDFLAGS` to makefile invocations? The alternative is to internally update `CEED_LDFLAGS` and let users customize `LDFLAGS`
```
Makefile:# Allow users to customize LDFLAGS (libceed appends directly to this variable)
Makefile:override LDFLAGS := $(LDFLAGS)
Makefile:  LDFLAGS += --coverage
Makefile:LDFLAGS += $(if $(ASAN),$(AFLAGS))
Makefile:       $(info LDFLAGS       = $(value LDFLAGS))
Makefile:$(libceed.so) : LDFLAGS += $(if $(DARWIN), -install_name @rpath/$(notdir $(libceed.so)))
Makefile:$(libceeds) : LDFLAGS += $(_pkg_ldflags) $(_pkg_ldflags:-L%=-Wl,-rpath,%) $(PKG_STUBS_LIBS)
Makefile:$(examples) $(tests) : LDFLAGS += $(_pkg_ldflags) $(_pkg_ldflags:-L%=-Wl,-rpath,%) $(PKG_STUBS_LIBS)
Makefile:       $(call quiet,LINK) $(LDFLAGS) -shared -o $@ $^ $(LDLIBS)
Makefile:       $(call quiet,LINK.c) $(CEED_LDFLAGS) -o $@ $(abspath $<) $(CEED_LIBS) $(LDLIBS)
Makefile:       $(call quiet,LINK.F) -DSOURCE_DIR='"$(abspath $(<D))/"' $(CEED_LDFLAGS) -o $@ $(abspath $<) $(CEED_LIBS) $(LDLIBS)
Makefile:       $(call quiet,LINK.c) $(CEED_LDFLAGS) -o $@ $(abspath $<) $(CEED_LIBS) $(LDLIBS)
Makefile:       $(call quiet,LINK.F) -DSOURCE_DIR='"$(abspath $(<D))/"' $(CEED_LDFLAGS) -o $@ $(abspath $<) $(CEED_LIBS) $(LDLIBS)
Makefile:$(tests) $(examples) : LDFLAGS += -Wl,-rpath,$(abspath $(LIBDIR)) -L$(LIBDIR)
Makefile:       AR ARFLAGS LDFLAGS LDLIBS LIBCXX SED \
examples/ceed/Makefile: $(LINK.c) $(CEED_FLAGS) $(CEED_LDFLAGS) $< -o $@ $(CEED_LIBS)
examples/fluids/Makefile:LDFLAGS = $(call pkgconf, --libs-only-L --libs-only-other $(PETSc.pc) $(ceed.pc))
examples/fluids/Makefile:LDFLAGS += $(patsubst -L%, $(call pkgconf, --variable=ldflag_rpath $(PETSc.pc))%, $(call pkgconf, --libs-only-L $(PETSc.pc) $(ceed.pc)))
examples/fluids/Makefile:       $(call quiet,LINK.o) $(CEED_LDFLAGS) $^ $(LOADLIBES) $(LDLIBS) -o $@
examples/fluids/Makefile:       $(info LDFLAGS : $(LDFLAGS))
examples/petsc/Makefile:LDFLAGS = $(call pkgconf, --libs-only-L --libs-only-other $(PETSc.pc) $(ceed.pc))
examples/petsc/Makefile:LDFLAGS += $(patsubst -L%, $(call pkgconf, --variable=ldflag_rpath $(PETSc.pc))%, $(call pkgconf, --libs-only-L $(PETSc.pc) $(ceed.pc)))
examples/petsc/Makefile:        $(call quiet,LINK.o) $(CEED_LDFLAGS) $^ $(LOADLIBES) $(LDLIBS) -o $@
examples/petsc/Makefile:        $(call quiet,LINK.o) $(CEED_LDFLAGS) $^ $(LOADLIBES) $(LDLIBS) -o $@
examples/petsc/Makefile:        $(call quiet,LINK.o) $(CEED_LDFLAGS) $^ $(LOADLIBES) $(LDLIBS) -o $@
examples/petsc/Makefile:        $(call quiet,LINK.o) $(CEED_LDFLAGS) $^ $(LOADLIBES) $(LDLIBS) -o $@
examples/petsc/Makefile:        $(call quiet,LINK.o) $(CEED_LDFLAGS) $^ $(LOADLIBES) $(LDLIBS) -o $@
examples/petsc/Makefile:        $(info LDFLAGS : $(LDFLAGS))
examples/solids/Makefile:LDFLAGS = $(call pkgconf, --libs-only-L --libs-only-other $(PETSc.pc) $(ceed.pc))
examples/solids/Makefile:LDFLAGS += $(patsubst -L%, $(call pkgconf, --variable=ldflag_rpath $(PETSc.pc))%, $(call pkgconf, --libs-only-L $(PETSc.pc) $(ceed.pc)))
examples/solids/Makefile:       $(call quiet,LINK.o) $(CEED_LDFLAGS) $^ $(LOADLIBES) $(LDLIBS) -o $@
examples/solids/Makefile:       $(info LDFLAGS : $(LDFLAGS))

```

Reply:
Thanks for pushing an updated makefile.  I will test it as soon as as I have time.  One suggestion.  I would swap LDFLAGS and CEED_LDFLAGS for the linking rule, so that users will always have the last word. (Last takes precedence)

Reply:
I was thinking about that, but I think the order now is what you want. `LDFLAGS` like `-L` are searched in the order specified on the command line so the last position is only reached if the library isn't found in any of the other directories. This is opposite flags like `-O`, in which the last occurrence has precedence.

You can see the precedence in action with this experiment.
```console
$ echo 'int main() { return 0;}' > a.c
$ gcc a.c -L/opt/mpich/lib -L/usr/lib/openmpi -lmpi
$ lddtree a.out    # or ldd or libtree
a.out (interpreter => /lib64/ld-linux-x86-64.so.2)
    libmpi.so.12 => /opt/mpich/lib/libmpi.so.12
        libhwloc.so.15 => /usr/lib/libhwloc.so.15
            libudev.so.1 => /usr/lib/libudev.so.1
                libgcc_s.so.1 => /usr/lib/libgcc_s.so.1
        libucp.so.0 => /opt/mpich/lib/libucp.so.0
        libuct.so.0 => /opt/mpich/lib/libuct.so.0
        libucs.so.0 => /opt/mpich/lib/libucs.so.0
        libnuma.so.1 => /usr/lib/libnuma.so.1
        libm.so.6 => /usr/lib/libm.so.6
        libucm.so.0 => /opt/mpich/lib/libucm.so.0
        libpthread.so.0 => /usr/lib/libpthread.so.0
        libz.so.1 => /usr/lib/libz.so.1
        libdl.so.2 => /usr/lib/libdl.so.2
        librt.so.1 => /usr/lib/librt.so.1
    libc.so.6 => /usr/lib/libc.so.6
```